### PR TITLE
Add mapbox and open street map attribution to report builder map

### PR DESCRIPTION
**Pull request by gherceg**
[JIRA ticket](https://dimagi-dev.atlassian.net/browse/SAAS-11135)
##### SUMMARY

When using MapBox and/or Open Street Map, we are required to add the correct attributions to maps we display. As pointed out by Kevin, attributions were previously being set on each layer [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/reports/static/reports/js/maps_utils.js#L348-L373) but it appears that code is not called on the report builder page. Ideally will look into cleaning up maps_utils.js in the future, but for the purpose of this JIRA ticket this PR contains only the changes needed.

To simplify attributions in relation to what layer (street or satellite) is being displayed, all attributions are displayed regardless of the displayed layer. The mapbox wordmark was obtained from [this link](https://docs.mapbox.com/help/how-mapbox-works/attribution/#other-mapping-frameworks) which specifically references properly attributing when using Leaflet.

### CODE REVIEW

**Comment by knguyen142** [comment]
@gherceg can you add screenshots of what it looks like (before and after)?

**Comment by gherceg** [comment]
@knguyen142 good point. Here's the before and after. Open to other suggestions for where the scale control should go. 

**Before**
Screen Shot 2020-09-01 at 8 31 31 AM
**After**
Screen Shot 2020-09-01 at 8 28 20 AM

**Review by millerdev — APPROVED** [review]
I'm not very familiar with this code, so other reviewer's feedback may be more important than mine.

File: corehq/apps/reports_core/static/reports_core/js/maps.js [hunk 1]
```diff
@@ -33,6 +33,12 @@ hqDefine('reports_core/js/maps', function () {
             privates.layerControl.addTo(privates.map);
 
             new (hqImport("reports/js/maps_utils").ZoomToFitControl)().addTo(privates.map);
+            // Add MapBox wordmark and correct attributes to map
+            // See https://docs.mapbox.com/help/how-mapbox-works/attribution/
+            new (hqImport("reports/js/maps_utils").MapBoxWordMark)().addTo(privates.map);
+            // scale is now placed on the bottom right because it is easier to layout with the attributes than with the wordmark
+            L.control.attribution({position: 'bottomright'}).addAttribution('&copy; <a href="http://www.mapbox.com/about/maps/">MapBox</a> | &copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a> | <strong><a href="https://www.mapbox.com/map-feedback/" target="_blank">Improve this map</a></strong>').addTo(privates.map);
```
Comment (knguyen142) [inline corehq/apps/reports_core/static/reports_core/js/maps.js:40]: I don't think the `Improve this map` is needed so I had it removed here https://github.com/dimagi/commcare-hq/blob/f8f29068699aeadd0d8b0cf488acec5150926b0e/corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js#L888-L889

**Review by knguyen142 — APPROVED** [review]
For some reason I missed that the mapbox wordmark was needed when creating https://github.com/dimagi/commcare-hq/blob/f8f29068699aeadd0d8b0cf488acec5150926b0e/corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js#L884-L898

Screen Shot 2020-09-02 at 5 11 29 PM

We can add it in a separate PR, but is there a way to refactor this to easily share the wordmark between the 2?

File: corehq/apps/reports_core/static/reports_core/js/maps.js (hunk 1 above)
Comment (gherceg) [reply corehq/apps/reports_core/static/reports_core/js/maps.js]: Removed the `Improve this map` part, and will handle adding the wordmark on a separate PR.

**Pull request merged**